app/services/ExactTargetService.scala

```diff
@@ -261,8 +270,10 @@ object SqsClient extends LazyLogging {
             "SubscriberAttributes" ->  Json.toJsFieldJsValueWrapper(row.fields.toMap)
           )
         ),
-        "DataExtensionName" -> row.forExtension.name
-      ).toString()
+        "DataExtensionName" -> row.forExtension.name,
+        "UserId" -> userId,
+        "SfContactId" -> sfContactId
```

Comment:
Having the SfContactId in the data might actually help to speed up the Contact lookup in the workflow service. I think it was searching for one by Identity ID. I'll set a reminder to have a look into that.

Reply:
Actually, you could pick up the SF Contact ID in the workflow service, and not need to change subs frontend?